apps/users-api/src/routes/registerUser.ts

```diff
@@ -0,0 +1,53 @@
+import { Router } from 'express';
+import { z } from 'zod';
+import bcrypt from 'bcrypt';
+
+import { prisma } from '@competition-manager/prisma';
+import { parseRequest, generateAccessToken, generateRefreshToken } from '@competition-manager/utils';
+
+export const router = Router();
+
+const Body$ = z.object({
+    email: z.string({ message: 'Email must be a string' }).email({ message: 'Email must be a valid email' }),
+    password: z.string({ message: 'Password must be a string' }).min(8, { message: 'Password must be at least 8 characters long' })
+});
+
+router.post(
+    '/register',
+    parseRequest('body', Body$),
+    async (req, res) => {
+        const { email, password } = Body$.parse(req.body);
+        const user = await prisma.user.findUnique({
+            where: {
+                email: email
+            }
+        });
+        if (user) {
+            res.status(409).json({ message: 'Email already in use' });
+            return;
+        } 
+        const hashedPassword = await bcrypt.hash(password, 10);
+        await prisma.user.create({
```

Comment:
est ce qu'on peut récup le user de create ?
```suggestion
        const user = await prisma.user.create({
```